### PR TITLE
Cancel Command execution

### PR DESCRIPTION
**Pull request by colombod**
New scheduler logic
New kernel routing
Implement cancel logic

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.Tests/Utility/CancellableCommand.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading.Tasks;
+using Microsoft.DotNet.Interactive.Commands;
+
+namespace Microsoft.DotNet.Interactive.Tests.Utility
+{
+    public class CancellableCommand : KernelCommand
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Tests/Utility/CancellableCommand.cs:9]: Is this being used?

File: src/Microsoft.DotNet.Interactive/Commands/KernelCommand.cs [hunk 2]
```diff
@@ -40,6 +40,9 @@ protected KernelCommand(string targetKernelName = null, KernelCommand parent = n
 
         internal static KernelCommand None { get; } = new NoCommand();
 
+        [JsonIgnore]
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/Commands/KernelCommand.cs:43]: The `[JsonIgnore]` shouldn't be needed for `internal` properties.

File: src/Microsoft.DotNet.Interactive/CompositeKernel.cs [hunk 3]
```diff
@@ -202,22 +203,42 @@ protected override void SetHandlingKernel(KernelCommand command, KernelInvocatio
...
+            var targetKernelName = command switch
+            {
+                { } kcb => kcb.TargetKernelName ?? DefaultKernelName,
+                _ => DefaultKernelName
+            };
+
+            Kernel kernel;
 
-            if (kernel is null)
+            if (targetKernelName != null)
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/CompositeKernel.cs:217]:
```suggestion
            if (targetKernelName is not null)
```

Comment (colombod) [reply src/Microsoft.DotNet.Interactive/CompositeKernel.cs:217]: done

**Pull request closed without merging**